apicurio-registry.yaml

```diff
@@ -0,0 +1,119 @@
+package:
+  name: apicurio-registry
+  version: 3.0.4
+  epoch: 0
+  description: An API/Schema registry - stores APIs and Schemas
+  copyright:
+    - license: Apache-2.0
+  target-architecture:
+    - x86_64 # Currently, we can't build for aarch64: https://github.com/Apicurio/apicurio-registry/issues/5633
+
+environment:
+  contents:
+    packages:
+      - bash
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - curl
+      - go
+      - icu
+      - maven
+      - nodejs-20
+      - npm
+      - openjdk-17
+      - openjdk-17-default-jvm
+      - openssf-compiler-options
+  environment:
+    JAVA_HOME: /usr/lib/jvm/java-17-openjdk
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/Apicurio/apicurio-registry
+      tag: v${{package.version}}
+      expected-commit: f417d2192cdef84dc7587842d98ed721dc3901e3
+
+  - uses: patch
+    with:
+      patches: CVE-2024-31141.patch
+
+  - name: Build
+    runs: |
+      ./mvnw clean install \
+        -Pprod \
+        -DskipTests \
+        -T$(nproc)C \
+        --no-snapshot-updates \
+        --no-transfer-progress \
+        --fail-fast
+
+  - name: Install
+    runs: |
+      dest="${{targets.contextdir}}/usr/share/java/${{package.name}}"
+      mkdir -p "$dest"
+
+      install -Dm644 ./LICENSE "${dest}/LICENSE"
+      install -Dm644 ./README.md "${dest}/README.md"
+
+      tar -zxf ./app/target/apicurio-registry-app-${{package.version}}-all.tar.gz -C "$dest"
+      find . -type f -path "*/target/*-${{package.version}}.jar" -exec cp {} "$dest" \;
+      find . -type f -path "*/target/*-${{package.version}}-runner.jar" -exec cp {} "$dest" \;
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-ui
+    description: Web UI for Apicurio Registry (apicurio-registry-ui)
+    dependencies:
+      runtime:
+        - nodejs-20
+    pipeline:
+      - working-directory: ui
+        runs: |
+          npm install
+          npm audit fix --package-lock-only --legacy-peer-deps || true
+          npm run build
+
+          mkdir -p ${{targets.contextdir}}/opt/app-root/src
+          cp -r ui-app/dist/* ${{targets.contextdir}}/opt/app-root/src/
```

Comment:
copying complete dist also copies a lot of `*.dll` files which is windows specific, just curious if we can avoid that or is that intentional to bundle as part of this package. 💭 

```bash
-rwxr--r-- root/root     35328 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/DotNet.Glob.dll
-rwxr--r-- root/root   1430048 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Kiota.Builder.dll
-rwxr--r-- root/root    308896 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.CSharp.dll
-rwxr--r-- root/root     64288 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Extensions.DependencyInjection.Abstractions.dll
-rwxr--r-- root/root     65288 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Extensions.Logging.Abstractions.dll
-rwxr--r-- root/root     90656 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Kiota.Abstractions.dll
-rwxr--r-- root/root     97832 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Kiota.Http.HttpClientLibrary.dll
-rwxr--r-- root/root     35376 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Kiota.Serialization.Form.dll
-rwxr--r-- root/root     66080 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Kiota.Serialization.Json.dll
-rwxr--r-- root/root     20528 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Kiota.Serialization.Multipart.dll
-rwxr--r-- root/root     27168 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.Kiota.Serialization.Text.dll
-rwxr--r-- root/root     58400 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.OpenApi.ApiManifest.dll
-rwxr--r-- root/root    146488 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.OpenApi.Readers.dll
-rwxr--r-- root/root    243256 2024-12-01 18:57:06 opt/app-root/src/kiota-wasm/Microsoft.OpenApi.dll
```
On size front also, this'll greatly reduce the size of the package.

Reply:
Nice catch! You should able to run native languages when using the WASM, so those should be OK, notice the `kiota-wasm`.